interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php

```diff
@@ -55,11 +65,66 @@ public function __construct()
      * @param $options
      * @return void
      */
-    public function setOptions($options)
+    public function setOptions($pid, $encounter, $options)
     {
-        $this->searchFromDate = !empty($options['date_start'] ?? null) ? date("Y-m-d H:i:s", strtotime($options['date_start'])) : null;
-        $this->searchToDate = !empty($options['date_end'] ?? null) ? date("Y-m-d H:i:s", strtotime($options['date_end'])) : null;
-        $this->searchFiltered = !empty($options['filter_content'] ?? false);
+        // we keep from and to dates in order to handle the transaction table where we have to manually convert the dates
+        // since they are stored as strings.
+        $this->searchFromDate = null;
+        $this->searchToDate = null;
+        $this->encounterFilterList = [];
+
+        $dateValues = [];
+
+        if (!empty($options['date_start'])) {
+            $searchFromDate = strtotime($options['date_start']);
```

Comment:
This concerns me a bit. Should this be coalesced to null. May be okay as I think $options is qualified before here unless there is another path calling this class from elsewhere, portal maybe.
Dates always get me a flutter but I trust you and will help test them once merged.
Looks good to me and thanks for the tedious work. Oh to be young again!:)